src/node.h

```diff
@@ -633,7 +633,18 @@ inline void NODE_SET_PROTOTYPE_METHOD(v8::Local<v8::FunctionTemplate> recv,
 #define NODE_SET_PROTOTYPE_METHOD node::NODE_SET_PROTOTYPE_METHOD
 
 // BINARY is a deprecated alias of LATIN1.
-enum encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER, LATIN1 = BINARY};
+// BASE64URL is not currently exposed to the JavaScript side.
+enum encoding {
+  ASCII,
+  UTF8,
+  BASE64,
+  BASE64URL,
```

Comment:
This is an ABI-breaking change. I would recommend adding this value at the end, even though that’s not quite as pretty.

Reply:
Yeah.. unfortunately because that's a bit messy because of the `LATIN1 = BINARY` ... I'm not 100% clear, is the following ABI breaking also or do we have to put BASE64URL after LATIN1?

```cpp
// Can we do this without breaking ABI...
enum encoding {
  ASCII,
  UTF8,
  BASE64,
  UCS2,
  BINARY,
  HEX,
  BUFFER,
  BASE64URL,
  LATIN1 = BINARY,
};
```

```cpp
// Or does it have to be this
enum encoding {
  ASCII,
  UTF8,
  BASE64,
  UCS2,
  BINARY,
  HEX,
  BUFFER,
  LATIN1 = BINARY,
  // Because LATIN1 is set explicitly to equal BINARY, we
  // have to explicitly set BASE64URL to a value to continue
  // on, otherwise it will be set to BINARY + 1...
  BASE64URL = 7
};
```

Reply:
The former option should work, yes :)